be/src/util/compression/compression_core_local_context_pool.h

```diff
@@ -0,0 +1,136 @@
+// This file is made available under Elastic License 2.0.
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "common/status.h"
+#include "common/statusor.h"
+#include "util/compression/compression_context_pool.h"
+#include "util/cpu_info.h"
+
+namespace starrocks::compression {
+
+/**
+ * This class is intended to reduce contention on reserving a compression
+ * context and improve cache locality (but maybe not hotness) of the contexts
+ * it manages.
+ *
+ * This class uses CpuInfo::get_current_core() to spread the managed object across
+ * NumStripes domains (which should correspond to a topologically close set of
+ * hardware threads). This cache is still backed by the basic locked stack in
+ * the compression::CompressionContextPool.
+ *
+ * Note that there is a tradeoff in choosing the number of stripes. More stripes
+ * make for less contention, but mean that a context is less likely to be hot
+ * in cache.
+ */
+template <typename T, typename Creator, typename Deleter, typename Resetter, size_t NumStripes = 32>
+class CompressionCoreLocalContextPool {
```

Comment:
Our threads are all created in advance. Is it meaningless to do another core-related one, or even conflict?

Reply:
But we have a small number of threads, and there is only one thread per thread. I feel that thread local is enough.

Reply:
i remove the core-local-cache compression pool. 
Because flush threads are created from the threads pool, if i use the thread local compression context, the context will still be frequently released and allocated when flushing thread is switched, so here i didn't use the thread local.